### DISCUSSION TITLE
Set a custom shape with views?

### DISCUSSION THREAD

**Discussion by csbenz** [patrykandpatrick/vico#532] [Questions]
I'm trying to change the shape of my column drawables, but the shape property is not reassignable. Is this the correct way to do it?

```
with(cartesianChartView.chart?.layers?.get(0) as ColumnCartesianLayer) {
            columns[0].color = ContextCompat.getColor(context, R.color.primary)
            columns[1].color = ContextCompat.getColor(context, R.color.secondary)
            mergeMode = { ColumnCartesianLayer.MergeMode.Stacked }
            spacingDp = 5f
            columns[0].shape = Shapes.roundedCornerShape(50)
        }
```

**Comment by Gowsky** [comment]
Hi @csbenz, `LineComponent.shape` (along with other properties) currently cannot be reassigned, and we'll be looking into making the API more flexible.
However, I'd recommend you to update  `ColumnCartesianLayer.columns` entirely with your list of `LineComponent`s, instead of mutating their properties.
You can also change the shape directly through XML attributes like we did [here](https://github.com/patrykandpatrick/vico/blob/89d0a098126a33227a98c86c39d37ed186678208/sample/src/main/res/values/chart_6_styles.xml) for [`Chart6`](https://github.com/patrykandpatrick/vico/blob/89d0a098126a33227a98c86c39d37ed186678208/sample/src/main/res/layout/chart_6.xml). For more information refer to the [wiki](https://patrykandpatrick.com/vico/wiki/2.0.0-alpha.5/cartesian-charts/hosts#xml-attributes).
